src/server/pfs/server/master.go

```diff
@@ -216,9 +215,48 @@ func (d *driver) finishRepoCommits(ctx context.Context, repoKey string) error {
 				log.Error(ctx, "error finishing commit", zap.Error(err), zap.Duration("retryAfter", d))
 				return nil
 			})
-		}, zap.Bool("finishing", true), log.Proto("commit", commit), zap.String("repo", key))
+		}, zap.Bool("finishing", true), log.Proto("commit", commitInfo.Commit), zap.String("repo", key))
 	}, watch.WithSort(col.SortByCreateRevision, col.SortAscend), watch.IgnoreDelete)
-	return errors.EnsureStack(err)
+}
+
+func (d *driver) compactDiffFileset(ctx context.Context, commit *pfs.Commit, compactor *compactor, renewer *fileset.Renewer, doer task.Doer) error {
```

Comment:
nit, would you change the ordering to ctx, compactor, taskDoer, renewer, commit. Ordering by scope generally reads better imo. Also, would you do this for the total file set method as well.